internal/controller/apiextensions/composite/composition_functions.go

```diff
@@ -133,6 +147,19 @@ func (fn ComposedResourceObserverFn) ObserveComposedResources(ctx context.Contex
 	return fn(ctx, xr)
 }
 
+// A ExtraResourcesGetter gets extra resources matching a selector.
+type ExtraResourcesGetter interface {
+	Get(ctx context.Context, selector *v1beta1.ResourceSelector) (*v1beta1.Resources, error)
```

Comment:
```suggestion
	Get(ctx context.Context, rs *v1beta1.ResourceSelector) (*v1beta1.Resources, error)
```

Nit: `selector` is redundant given the type.